### PR TITLE
fix: patch latest & topic on desktop channel

### PR DESCRIPTION
**Pull request by Aylie-Chou**
This patch fix defect [[defect] header 未顯示「最新」、「深度專題」兩個項目](https://app.asana.com/0/1203699264568808/1203964076256284/f)

### CODE REVIEW

File: packages/universal-header/src/utils/links.js [hunk 1]
```diff
@@ -190,8 +190,19 @@ export function getSocialMediaLinks() {
...
-  releaseBranch = defaultReleaseBranch
+  releaseBranch = defaultReleaseBranch,
+  targetChannelKey = ''
 ) {
+  if (targetChannelKey) {
+    return __getLink(
+      isExternal,
+      releaseBranch,
+      mainBaseURL,
+      CHANNEL_PATH[targetChannelKey]
```
Comment (duidae) [inline packages/universal-header/src/utils/links.js:201]: we may need error handling for `CHANNEL_PATH[targetChannelKey]`

**Review by duidae — CHANGES_REQUESTED** [review]

File: packages/universal-header/src/utils/links.js (hunk 1 above)
Comment (Aylie-Chou) [reply packages/universal-header/src/utils/links.js]: error handle for parameter `targetChannelKey` added ([cbda297](https://github.com/twreporter/twreporter-npm-packages/pull/369/commits/cbda2974b2b9f54f9d3019aa150fad9bfef46774))
also seperate `getChannelKey` & `getChannelKeys` function.

File: packages/universal-header/src/utils/links.js [hunk 2]
```diff
@@ -188,6 +188,23 @@ export function getSocialMediaLinks() {
   )
 }
 
+export function getChannelLink(
+  isExternal = defaultIsExternal,
+  releaseBranch = defaultReleaseBranch,
+  targetChannelKey = ''
+) {
+  if (!targetChannelKey || !CHANNEL_PATH[targetChannelKey]) {
+    throw new Error('invalid channel key')
```
Comment (duidae) [inline packages/universal-header/src/utils/links.js:197]: here we throw the exception but not catching it, seems does not have the effect of error handling, could just copy other js function's handling (ex: find() of array)), return undefined and check the returned value before use

**Review by duidae — CHANGES_REQUESTED** [review]

File: packages/universal-header/src/utils/links.js (hunk 2 above)
Comment (Aylie-Chou) [reply packages/universal-header/src/utils/links.js:197]: The parameter `targetChannelKey` would basically use values setting in channel constant file, and would not pass any value from user input, thus I think the throwing error is for the development use.
In this scenario, current code would throw error in twreporter-react SSR. I think it supply enough info for developer to understand where went run.

Comment (duidae) [reply packages/universal-header/src/utils/links.js:197]: once we give `getChannelLink` open access to everyone, developers would possibly put anything in the arguments and we may not assume people to know any implicit rule when calling it. We either pull `CHANNEL_PATH` out of this function for clear or properly handle parameters/return value when doing a function call.

**Review by duidae — CHANGES_REQUESTED** [review]

File: packages/universal-header/src/utils/links.js (hunk 2 above)
Comment (Aylie-Chou) [reply packages/universal-header/src/utils/links.js]: updated ([c7856a1](https://github.com/twreporter/twreporter-npm-packages/pull/369/commits/c7856a10811b9053577bcde36aeb73a69c007784))

- add `getLink` to replace `getChannelLink` & `getCategoryLink` function
- add error handle in Channel component

**Review by duidae — APPROVED** [review]

**Pull request merged**